src/Util/AutoTimeoutUnsubscriber/index.js

```diff
@@ -19,3 +19,14 @@ export default function createNewInstanceOfAutoTimeoutUnsubscribe() {
     }
   };
 }
+
+export function PromiseWithTimeout(promises, timeout) {
+  let promise = promises;
+  if (promises instanceof Array) promise = Promise.all(promises);
```

Comment:
BTW, `Promise.allSettled` might be better than `Promise.all` in this case. Depends on your preference.
@Jakub41